packages/ai/core/generate-text/smooth-stream.ts

```diff
@@ -60,10 +77,13 @@ export function smoothStream<TOOLS extends ToolSet>({
         buffer += chunk.textDelta;
 
         let match;
-        while ((match = chunkingRegexp.exec(buffer)) != null) {
-          const chunk = match[0];
-          controller.enqueue({ type: 'text-delta', textDelta: chunk });
-          buffer = buffer.slice(chunk.length);
+
+        while (
+          typeof (match = detectChunk(buffer)) === 'string' &&
```

Comment:
added more validation to ensure the user passes correct stuff

Reply:
prefer to revert - it hides issues on the users side that they need to fix and adds complexity

Reply:
i've improved it to now throw an error